UVA-Relay/Program.cs

```diff
@@ -1,39 +1,146 @@
 ﻿using System;
+using Tomlet;
+using System.Collections.Generic;
+using System.IO;
 using System.Threading.Tasks;
 using DSharpPlus;
 using DSharpPlus.Interactivity;
 using DSharpPlus.Interactivity.Enums;
 using DSharpPlus.Interactivity.Extensions;
 using DSharpPlus.SlashCommands;
+using Tomlet.Models;
 using UVACanvasAccess.ApiParts;
+using UVACanvasAccess.Util;
+using UVA_Relay.sql;
 
 namespace UVA_Relay {
     internal static class Program {
-        public static async Task Main() {
-            Canvas.CanvasApi = new Api(Environment.GetEnvironmentVariable("UVA_RELAY_CANVAS_TOKEN"),
-                                       "https://uview.instructure.com/api/v1/");
+
+        public static async Task Main()
+        {
+            //SQL initialize
+            SQL db = new SQL();
+            // Get tokens from toml file
+            TomlDocument document = TomlParser.ParseFile(Environment.GetEnvironmentVariable("config"));
+            // document.GetSubTable("keys").GetString("BOT_TOKEN"));
+            Canvas.CanvasApi = new Api(document.GetSubTable("keys").GetString("UVA_RELAY_CANVAS_TOKEN"),
+                "https://uview.instructure.com/api/v1/");
             
             var discord = new DiscordClient(new DiscordConfiguration {
-                Token = Environment.GetEnvironmentVariable("UVA_RELAY_TOKEN") ?? throw new NullReferenceException("UVA_RELAY_TOKEN is unset"),
+                //Token = Environment.GetEnvironmentVariable("UVA_RELAY_TOKEN") ?? throw new NullReferenceException("UVA_RELAY_TOKEN is unset"),
+                Token = document.GetSubTable("keys").GetString("BOT_TOKEN"),
                 TokenType = TokenType.Bot,
                 Intents = DiscordIntents.All
             });
 
             await discord.ConnectAsync();
 
             discord.UseInteractivity(new InteractivityConfiguration {
-                AckPaginationButtons = true,
                 PaginationBehaviour = PaginationBehaviour.Ignore
             });
-
+            /*
             var testGuildId = ulong.Parse(Environment.GetEnvironmentVariable("UVA_RELAY_TEST_GUILD_ID")
                                           ?? throw new NullReferenceException("UVA_RELAY_TEST_GUILD_ID is unset"));
-            
+            */
+            var testGuildId = ulong.Parse(document.GetSubTable("keys").GetString("GUILD_ID")
+                                          ?? throw new NullReferenceException("UVA_RELAY_TEST_GUILD_ID is unset"));
             var slash = discord.UseSlashCommands();
+            
+            //Slash commands error handler for displaying cooldown and such
+            slash.SlashCommandErrored += Utils.CmdErroredHandler;
+
+            // Get all the guilds the bot is in when starting program
+            discord.GuildDownloadCompleted += (s, e) =>
```

Comment:
This looks like it is adding every user in every guild to the database upon startup. This shouldn't be needed -- there will be an explicit pairing process to associate a Canvas user with a Discord user. We probably do need to do this for guilds themselves, so keep that.